simcore/simulation/library/anchor.cpp

```diff
@@ -195,6 +197,20 @@ void Anchor::UpdateAnchorPositionToBond() {
   }
   UpdatePeriodic();
 }
+/*Creates Vector that has different binding rates for parallel and anti-parallel bonds*/
+std::vector<double> Anchor::CreateOrientationArray(int n_neighbors){
+  double const *const orientation = bond_->GetOrientation();
+   std::vector<double> OrientationArray(n_neighbors, k_on_d_);     
+   for(int j=0; j<n_neighbors;++j){
+       	Object* obj=neighbors_.GetNeighbor(j);  
+        double const* const n_orientation=obj->GetOrientation();
+	double DotProduct=dot_product(n_dim_,orientation,n_orientation);
+        if (DotProduct<0){
+		OrientationArray[j]=k_on_d_*polar_affinity_;
+        }   	 
+   }
+   return OrientationArray;
```

Comment:
This function should not need to return a vector. The function that calls it already has a vector as you defined it above. So anchor should not need to know k_on_d_ or n_neighbors (that would come from the vector size). Instead, you should have passed the existing vector by reference and changed their values.